### PR TITLE
[MXNET-309] [ONNX-MXNet] Model Metadata API

### PR DESCRIPTION
**Pull request by anirudhacharya**
## Description ##
A new API in the onnx module to fetch shape and name information of a model.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- A new API to get name and shape information of an ONNX model

## Comments ##
- @spidyDev @Roshrini @sandeep-krishnamurthy please review

### CODE REVIEW

**Comment by ThomasDelteil** [comment]
@anirudhacharya 
not sure, currently it looks like to find the input name you are doing this:
```
data_names = [graph_input for graph_input in sym.list_inputs()
                      if graph_input not in arg_params and graph_input not in aux_params]
print(data_names)
```
and that your new API would let it infer it automagically?

to replace this
```
net = gluon.nn.SymbolBlock(outputs=sym, inputs=mx.sym.var('gpu_0/data_0'))
```
with some deterministic API call?

The two tutorial I wrote would both need updating for model loading and datanames with the new API

**Comment by anirudhacharya** [comment]
@anirudh2290 pls review.

File: python/mxnet/contrib/onnx/_import/import_onnx.py [hunk 1]
```diff
@@ -132,6 +132,33 @@ def from_onnx(self, graph):
...
+        _params = set()
+        for tensor_vals in graph.initializer:
+            _params.add(tensor_vals.name)
+
+        input_data = []
+        for graph_input in graph.input:
+            shape = []
+            if graph_input.name not in _params:
+                for val in graph_input.type.tensor_type.shape.dim:
+                    shape.append(val.dim_value)
```
Comment (anirudh2290) [inline python/mxnet/contrib/onnx/_import/import_onnx.py]: shape = [val.dim_value for val in graph_input.type.tensor_type.shape.dim]

File: python/mxnet/contrib/onnx/_import/import_onnx.py [hunk 2]
```diff
@@ -132,6 +132,33 @@ def from_onnx(self, graph):
...
+                input_data.append((graph_input.name, tuple(shape)))
+
+        output_data = []
+        for graph_out in graph.output:
+            shape = []
```
Comment (anirudh2290) [inline python/mxnet/contrib/onnx/_import/import_onnx.py]: same as above

File: python/mxnet/contrib/onnx/_import/import_onnx.py (hunk 1 above)
Comment (anirudh2290) [reply python/mxnet/contrib/onnx/_import/import_onnx.py]: are all these attrs `type` `tensor_type` `shape` `dim` guaranteed to be not `None` if graph_input is not `None` ?

Comment (anirudhacharya) [reply python/mxnet/contrib/onnx/_import/import_onnx.py]: yes, it is part of the onnx graph definition. But I will put a new check and throw a model malformed exception here.

**Pull request merged**